components/scream/CMakeLists.txt

```diff
@@ -192,6 +197,10 @@ set(SCREAM_LIB_ONLY ${DEFAULT_LIB_ONLY} CACHE BOOL "Only build libraries, no exe
 set(NetCDF_Fortran_PATH ${DEFAULT_NetCDF_Fortran_PATH} CACHE FILEPATH "Path to netcdf fortran installation")
 set(NetCDF_C_PATH ${DEFAULT_NetCDF_C_PATH} CACHE FILEPATH "Path to netcdf C installation")
 set(SCREAM_MACHINE ${DEFAULT_SCREAM_MACHINE} CACHE STRING "The CIME/SCREAM name for the current machine")
+set(SCREAM_MONOLITHIC_KERNELS ${DEFAULT_MONOLITHIC_KERNELS} CACHE STRING "Use monolithic kokkos kernels")
+if (NOT SCREAM_MONOLITHIC_KERNELS)
+  set(EKAT_DISABLE_WORKSPACE_SHARING TRUE CACHE STRING "")
```

Comment:
I'm not 100% sure this is needed or even a good design choice. I think we should leave it to SCREAM to set up their WSMs properly. Any WSM that might be used for small kernels should have their over provision factor set to a huge number. This will prevent any workspace sharing. The top level functions of small kernels should assert that their WSMs have one slot per team.

Reply:
I agree. I'm not 100% sure what this option would do in EKAT. But I feel like we don't want to set _global_ ekat configurations just b/c _a few_ use cases need it. If we can make SCREAM create WSMs differently, each with its own config/ctor params, I think that would be better.

Reply:
EKAT's WSM should support a new option to make the workspace size proportional to league size (total number of physics columns) rather than use the overprovision factor. This will follow what Hommexx does.